perl_modules/Atlas/ZoomaClient.pm

```diff
@@ -244,6 +251,80 @@ sub _query_zooma {
 }
 
 
+sub zooma_ontology_lookup {
+    my ( $filename ) = @_;
+    my $abs_path = dirname(File::Spec->rel2abs(__FILE__));
+    my $zoomaOntologyLookupFile = "$abs_path/../../supporting_files/". $filename;
+    open (my $in_fh, '<', $zoomaOntologyLookupFile) or die $!;
+    my %ontology_lookup;
+    while ( my $line = <$in_fh> ) {
+        my ( $property_type, $organism, $ontologies ) = split /\t/, $line;
+        $ontology_lookup{$property_type}{$organism} = $ontologies;
+    }   
+    return \%ontology_lookup;
+}
+
+sub get_plants_species {
+    my ( $rnaseqer_api ) = @_;
+    my $plants_query=$rnaseqer_api . '/tsv/0/getOrganisms/plants';
+    my $ua = LWP::UserAgent->new;
+    my $request = HTTP::Request->new("GET" => $plants_query);
+    my $response = $ua->request($request);
+    my @plants_species_list;
+
+    if ( $response->is_success ) {
+        my $plants_results = $response->content_ref;
+        while (my $line = <${ $plants_results }>) {
+             next if $line =~ m/ORGANISM/;
+             my @plants_species = split(/\t/,$line);
+             push (@plants_species_list, $plants_species[0])
+        }
+        return \@plants_species_list;
+    }
+    else {
+        $logger->error(
+            "Problem querying RNASeq-er API to retrieve plant species: ",
+            $response->status_line,".");
+    }
+}
+
+
+sub get_ontology_for_type {
+    my ( $ontology_lookup, $organism, $property_type ) = @_;
+
+    # get plants gxa species from RNASeq-er API
+    my $rnaseqer_api='https://www.ebi.ac.uk/fg/rnaseq/api';
+    my $plant_species=get_plants_species( $rnaseqer_api ) ;
```

Comment:
Could we maybe do this call at higher level so that is not done on every single query for each term and we pass this species array or hash as an argument down the functions?